### PR TITLE
test: changes for test_annotation IT tests

### PR DESCRIPTION
**Pull request by asthamohta**
Adding changes in the following test cases:
1. TestIntegration_BatchDML_Error
2. TestIntegration_BatchDML_TwoStatements
3. TestIntegration_BatchDML_NoStatements
4. TestIntegration_BatchDML
5. TestIntegration_PDML
6. TestIntegration_DML
7. TestIntegration_BROTNormal
8. TestIntegration_BatchQuery
9. TestIntegration_ReadErrors
10. TestIntegration_TransactionRunner
11. TestIntegration_BatchRead
12. TestIntegration_InvalidDatabase
13. TestIntegration_QueryStats
14. TestIntegration_QueryExpressions
15. TestIntegration_DirectPathFallback
16. TestIntegration_StartBackupOperation

### CODE REVIEW

File: spanner/integration_test.go [hunk 1]
```diff
@@ -146,12 +156,28 @@ var (
 		`CREATE INDEX TestTableByValueDesc ON TestTable(StringValue DESC)`,
 	}
 
+	readDBPGStatements = []string{
+		`CREATE TABLE TestTable (
+                    Key          VARCHAR PRIMARY KEY,
+                    StringValue  VARCHAR
+            )`,
```
Comment (hengfengli) [inline spanner/integration_test.go:164]: the indentation looks a big weird here. Have we run any linter, like goimports, to format the code?

File: spanner/integration_test.go [hunk 2]
```diff
@@ -2698,10 +2755,29 @@ func TestIntegration_DML(t *testing.T) {
 		return got
 	}
 
+	singersQuery := [7]string{`INSERT INTO Singers (SingerId, FirstName, LastName) VALUES (1, "Umm", "Kulthum")`,
+		`Insert INTO Singers (SingerId, FirstName, LastName) VALUES (2, "Eduard", "Khil")`,
```
Comment (hengfengli) [inline spanner/integration_test.go]: `Insert` -> `INSERT`

File: spanner/integration_test.go [hunk 3]
```diff
@@ -2698,10 +2755,29 @@ func TestIntegration_DML(t *testing.T) {
...
+		`INSERT INTO Singers (SingerId, FirstName, LastName) VALUES (3, "Audra", "McDonald")`,
+		`UPDATE Singers SET FirstName = "Oum" WHERE SingerId = 1`,
+		`UPDATE Singers SET FirstName = "Eddie" WHERE SingerId = 2`,
+		`INSERT INTO Singers (SingerId, FirstName, LastName) VALUES (3, "Audra", "McDonald")`,
+		`SELECT FirstName from Singers`,
+	}
+
+	if testDialect == adminpb.DatabaseDialect_POSTGRESQL {
+		singersQuery = [7]string{`INSERT INTO Singers (SingerId, FirstName, LastName) VALUES (1, 'Umm', 'Kulthum')`,
+			`Insert INTO Singers (SingerId, FirstName, LastName) VALUES (2, 'Eduard', 'Khil')`,
```
Comment (hengfengli) [inline spanner/integration_test.go]: same here

File: spanner/integration_test.go [hunk 4]
```diff
@@ -2698,10 +2755,29 @@ func TestIntegration_DML(t *testing.T) {
 		return got
 	}
 
+	singersQuery := [7]string{`INSERT INTO Singers (SingerId, FirstName, LastName) VALUES (1, "Umm", "Kulthum")`,
+		`Insert INTO Singers (SingerId, FirstName, LastName) VALUES (2, "Eduard", "Khil")`,
+		`INSERT INTO Singers (SingerId, FirstName, LastName) VALUES (3, "Audra", "McDonald")`,
+		`UPDATE Singers SET FirstName = "Oum" WHERE SingerId = 1`,
+		`UPDATE Singers SET FirstName = "Eddie" WHERE SingerId = 2`,
+		`INSERT INTO Singers (SingerId, FirstName, LastName) VALUES (3, "Audra", "McDonald")`,
+		`SELECT FirstName from Singers`,
```
Comment (hengfengli) [inline spanner/integration_test.go]: from -> FROM

File: spanner/integration_test.go [hunk 5]
```diff
@@ -2698,10 +2755,29 @@ func TestIntegration_DML(t *testing.T) {
...
+	}
+
+	if testDialect == adminpb.DatabaseDialect_POSTGRESQL {
+		singersQuery = [7]string{`INSERT INTO Singers (SingerId, FirstName, LastName) VALUES (1, 'Umm', 'Kulthum')`,
+			`Insert INTO Singers (SingerId, FirstName, LastName) VALUES (2, 'Eduard', 'Khil')`,
+			`INSERT INTO Singers (SingerId, FirstName, LastName) VALUES (3, 'Audra', 'McDonald')`,
+			`UPDATE Singers SET FirstName = 'Oum' WHERE SingerId = 1`,
+			`UPDATE Singers SET FirstName = 'Eddie' WHERE SingerId = 2`,
+			`INSERT INTO Singers (SingerId, FirstName, LastName) VALUES (3, 'Audra', 'McDonald')`,
+			`SELECT FirstName from Singers`,
```
Comment (hengfengli) [inline spanner/integration_test.go]: from -> FROM

File: spanner/integration_test.go (hunk 1 above)
Comment (asthamohta) [reply spanner/integration_test.go:164]: It looks weird on github I think but on my system the last bracket is further ahead
Also I did run a linter

Comment (hengfengli) [reply spanner/integration_test.go:164]: okay, then it's fine.

File: spanner/integration_test.go [hunk 6]
```diff
@@ -3153,12 +3207,24 @@ func TestIntegration_BatchDML(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	singersQuery := [3]string{`UPDATE Singers SET Singers.FirstName = "changed 1" WHERE Singers.SingerId = 1`,
```
Comment (hengfengli) [inline spanner/integration_test.go]: I think you can write `[]string{...}` instead of `[3]string{...}`.

File: spanner/integration_test.go [hunk 7]
```diff
@@ -3232,19 +3296,31 @@ func TestIntegration_BatchDML_TwoStatements(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	singersQuery := [4]string{`UPDATE Singers SET Singers.FirstName = "changed 1" WHERE Singers.SingerId = 1`,
```
Comment (hengfengli) [inline spanner/integration_test.go]: same here.

File: spanner/integration_test.go [hunk 8]
```diff
@@ -3232,19 +3296,31 @@ func TestIntegration_BatchDML_TwoStatements(t *testing.T) {
...
+		`UPDATE Singers SET Singers.FirstName = "changed 2" WHERE Singers.SingerId = 2`,
+		`UPDATE Singers SET Singers.FirstName = "changed 3" WHERE Singers.SingerId = 3`,
+		`UPDATE Singers SET Singers.FirstName = "changed 1" WHERE Singers.SingerId = 1`}
+
+	if testDialect == adminpb.DatabaseDialect_POSTGRESQL {
+		singersQuery = [4]string{`UPDATE Singers SET FirstName = 'changed 1' WHERE SingerId = 1`,
```
Comment (hengfengli) [inline spanner/integration_test.go]: same here.

File: spanner/integration_test.go [hunk 9]
```diff
@@ -3282,11 +3357,21 @@ func TestIntegration_BatchDML_Error(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	singersQuery := [3]string{`UPDATE Singers SET Singers.FirstName = "changed 1" WHERE Singers.SingerId = 1`,
```
Comment (hengfengli) [inline spanner/integration_test.go]: same here.

File: spanner/integration_test.go [hunk 10]
```diff
@@ -3282,11 +3357,21 @@ func TestIntegration_BatchDML_Error(t *testing.T) {
...
+		`some illegal statement`,
+		`UPDATE Singers SET Singers.FirstName = "changed 3" WHERE Singers.SingerId = 3`}
+
+	if testDialect == adminpb.DatabaseDialect_POSTGRESQL {
+		singersQuery = [3]string{`UPDATE Singers SET FirstName = 'changed 1' WHERE SingerId = 1`,
```
Comment (hengfengli) [inline spanner/integration_test.go]: same here

**Review by hengfengli — APPROVED** [review]
LGTM.

**Pull request merged**